From 2ebd37750b7acb494c1118fc8005ce54df0f99df Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Mon, 15 Mar 2021 12:35:11 +0100
Subject: [PATCH] Deduplicate "Establish Embassy" move costs.

Doing post "Establish Embassy" escape to a nearby city spends all remaining
MP. There is no point in also subtracting 1 move fragment for doing the
action.

Reported by Marko Lindqvist <cazfi74@gmail.com>

See osdn #41765
---
 data/civ2/effects.ruleset          | 10 +---------
 data/civ2civ3/effects.ruleset      | 10 +---------
 data/classic/effects.ruleset       | 10 +---------
 data/goldkeep/effects.ruleset      | 12 ++----------
 data/multiplayer/effects.ruleset   | 10 +---------
 data/sandbox/effects.ruleset       | 10 +---------
 data/webperimental/effects.ruleset | 10 +---------
 7 files changed, 8 insertions(+), 64 deletions(-)

diff --git a/data/civ2/effects.ruleset b/data/civ2/effects.ruleset
index d1f59a8be8..2cc3c5da92 100644
--- a/data/civ2/effects.ruleset
+++ b/data/civ2/effects.ruleset
@@ -2501,15 +2501,7 @@ reqs    =
       "Action", "Transport Embark", "Local", TRUE
     }
 
-[effect_action_success_move_cost_embassy]
-type    = "Action_Success_Actor_Move_Cost"
-value   = 1
-reqs    =
-    { "type", "name", "range", "present"
-      "Action", "Establish Embassy", "Local", TRUE
-    }
-
-[effect_action_success_establish_embassy]
+[effect_action_establish_embassy_success_move_cost]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 65535
 reqs    =
diff --git a/data/civ2civ3/effects.ruleset b/data/civ2civ3/effects.ruleset
index 2c0fe7e13e..ea2d7a6ce5 100644
--- a/data/civ2civ3/effects.ruleset
+++ b/data/civ2civ3/effects.ruleset
@@ -4251,7 +4251,7 @@ reqs    =
       "BuildingGenus", "GreatWonder", "Local"
     }
 
-[effect_action_success_establish_embassy]
+[effect_action_establish_embassy_success_move_cost]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 65535
 reqs    =
@@ -4331,14 +4331,6 @@ reqs    =
       "Action", "Sabotage Unit Escape", "Local", TRUE
     }
 
-[effect_action_success_move_cost_embassy]
-type    = "Action_Success_Actor_Move_Cost"
-value   = 1
-reqs    =
-    { "type", "name", "range", "present"
-      "Action", "Establish Embassy", "Local", TRUE
-    }
-
 [effect_action_success_move_cost_investigate]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 1
diff --git a/data/classic/effects.ruleset b/data/classic/effects.ruleset
index 3e5eb705b7..1c0b70ac9e 100644
--- a/data/classic/effects.ruleset
+++ b/data/classic/effects.ruleset
@@ -2524,7 +2524,7 @@ reqs    =
       "BuildingGenus", "GreatWonder", "Local"
     }
 
-[effect_action_success_establish_embassy]
+[effect_action_establish_embassy_success_move_cost]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 65535
 reqs    =
@@ -2604,14 +2604,6 @@ reqs    =
       "Action", "Sabotage Unit Escape", "Local", TRUE
     }
 
-[effect_action_success_move_cost_embassy]
-type    = "Action_Success_Actor_Move_Cost"
-value   = 1
-reqs    =
-    { "type", "name", "range", "present"
-      "Action", "Establish Embassy", "Local", TRUE
-    }
-
 [effect_action_success_move_cost_investigate]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 1
diff --git a/data/multiplayer/effects.ruleset b/data/multiplayer/effects.ruleset
index 45504390df..a36262de97 100644
--- a/data/multiplayer/effects.ruleset
+++ b/data/multiplayer/effects.ruleset
@@ -2569,14 +2569,6 @@ reqs    =
       "Building", "Palace", "Local", FALSE
     }
 
-[effect_action_success_move_cost_embassy]
-type    = "Action_Success_Actor_Move_Cost"
-value   = 1
-reqs    =
-    { "type", "name", "range", "present"
-      "Action", "Establish Embassy", "Local", TRUE
-    }
-
 [effect_action_success_move_cost_investigate]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 1
@@ -2630,7 +2622,7 @@ reqs    =
       "UnitFlag", "OneAttack", "Local", FALSE
     }
 
-[effect_action_success_establish_embassy]
+[effect_action_establish_embassy_success_move_cost]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 65535
 reqs    =
diff --git a/data/sandbox/effects.ruleset b/data/sandbox/effects.ruleset
index 114658dcbb..7b4a84dbae 100644
--- a/data/sandbox/effects.ruleset
+++ b/data/sandbox/effects.ruleset
@@ -4397,7 +4397,7 @@ reqs    =
       "BuildingGenus", "GreatWonder", "Local"
     }
 
-[effect_success_move_cost_establish_embassy]
+[effect_action_establish_embassy_success_move_cost]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 65535
 reqs    =
@@ -4493,14 +4493,6 @@ reqs    =
       "Action", "Steal Gold Escape", "Local", TRUE
     }
 
-[effect_action_success_move_cost_embassy]
-type    = "Action_Success_Actor_Move_Cost"
-value   = 1
-reqs    =
-    { "type", "name", "range", "present"
-      "Action", "Establish Embassy", "Local", TRUE
-    }
-
 [effect_action_success_move_cost_investigate]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 1
diff --git a/data/webperimental/effects.ruleset b/data/webperimental/effects.ruleset
index 7f96f79f1a..9cf44fefe2 100644
--- a/data/webperimental/effects.ruleset
+++ b/data/webperimental/effects.ruleset
@@ -2757,7 +2757,7 @@ reqs    =
       "BuildingGenus", "GreatWonder", "Local"
     }
 
-[effect_action_success_move_cost_establish_embassy]
+[effect_action_establish_embassy_success_move_cost]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 65535
 reqs    =
@@ -2861,14 +2861,6 @@ reqs    =
       "Action", "Suitcase Nuke Escape", "Local", TRUE
     }
 
-[effect_action_success_move_cost_embassy]
-type    = "Action_Success_Actor_Move_Cost"
-value   = 1
-reqs    =
-    { "type", "name", "range", "present"
-      "Action", "Establish Embassy", "Local", TRUE
-    }
-
 [effect_action_success_move_cost_investigate]
 type    = "Action_Success_Actor_Move_Cost"
 value   = 1
-- 
2.30.2

